adform-htb.js

```diff
@@ -144,9 +227,16 @@ function AdformHtb(configs) {
         var queryObj = {};
         var callbackId = System.generateUniqueId();
 
-        /* Change this to your bidder endpoint.*/
-        var baseUrl = Browser.getProtocol() + '//someAdapterEndpoint.com/bid';
+        var request = [];
+
+        for (var i in returnParcels) {
+            request.push(getRequestItem(returnParcels[i].xSlotRef));
+        }
 
+        request.unshift(Browser.getProtocol() + '//' + (configs.adxDomain || 'adx.adform.net') + '/adx/?rp=4&fd=1' + getQueryValue('tid', configs.tid) + getQueryValue('url', configs.url));
+
+        /* Change this to your bidder endpoint.*/
+        var baseUrl = request.join('&');
         /* ---------------- Craft bid request using the above returnParcels --------- */
 
 
```

Comment:
`getRequestItem` returns URI encoded values which later get appended to the `url` query parameter on the `baseUrl`. However, `getRequestItem` returns the same `mid` key multiple times. 

For example: `mid=654321` (encoded: bWlkPTY1NDMyMQ), `mid=54321` (encoded: bWlkPTY1NDMyMQ) & `mid=12345&mkw=keyword` (encoded: bWlkPTY1NDMyMQ). So the `url` paramater on baseUrl uses the `mid` key multiple times.

https://adx2.adform.net/adx/?rp=4&fd=1&tid=1&url=http%3A%2F%2Fadform.com&
`bWlkPTY1NDMyMQMyMQ`&`bWlkPTU0MzIx`&`bWlkPTEyMzQ1Jm1rdz1rZXl3b3Jk`

Was it intentional to provide the `url` query paramater with the same keys multiple times? 

Also, I noticed that you are not leverging the `queryObj` for  the query paramaters. Properties on this object will later be converted to a query string parameters in conjuction with the baseUrl. We would recommend using the `queryObj` for setting query string parameters.

Reply:
Yes, it's intentional to pass same parameter multiple times.

There are no clear rules defined how query string is generated from `queryObj`. If it's the same method as in `libraryStubData.js` then it doesn't fit our needs because it will generate `bWlkPTY1NDMyMQMyMQ=` instead of `bWlkPTY1NDMyMQMyMQ`. That's why only `baseUrl` is used.